graphs/lca.cpp

```diff
@@ -1,35 +1,39 @@
+#include <bits/stdc++.h>
+using namespace std;
 struct lca {
+	using pii = pair<int, int>;
 	int Log;
 	vector<vector<int>> memo;
-	vector<int> depth;
+	vector<int> depth, dist;
 
-	lca(const vector<vector<int>> &graph, int root) {//0 - based nodes
+	// use weights of 1 for unweighted tree
+	lca(const vector<vector<pii>> &graph, int root = 0) {//0 - based nodes
```

Comment:
let’s not have a default value for the root. I want to consciously be reminded to pass something in which I believe would reduce bugs:

Imagine if you do 1-based nodes and forgot to update the root to 1. If root defaults to zero, bug. If no default, compiler error, no bug.